dbsnap_verify/__init__.py

```diff
@@ -0,0 +1,176 @@
+)
+
+import boto3
+
+from sys import stdout
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.StreamHandler(stdout))
+
+
+def wait(state_doc, rds_session):
+    min_timestamp = state_doc["snapshot_minimum_timestamp"]
+    min_datetime = timestamp_to_datetime(min_timestamp)
+    logger.info("Looking for a snapshot of {} older then {}".format(
+        state_doc["database"],
+        datetime_to_date_str(min_datetime)
+    ))
+    snapshot_desc = get_latest_snapshot(rds_session, state_doc["database"])
+    if snapshot_desc["SnapshotCreateTime"] >= min_datetime:
+        # if the latest snapshot is older then the minimum, restore it.
+        restore(state_doc, rds_session)
+    elif min_datetime >= three_days_prior(now_datetime()):
+        # continue wating for a new snapshot, to restore.
+        transition_state(state_doc, "wait")
+        logger.info("Did not find a snapshot of {} older then {}".format(
+            state_doc["database"],
+            datetime_to_date_str(min_datetime)
+        ))
+        logger.info("Going to sleep.")
+    else:
+        logger.warning("Alert! we never found a snapshot of {} older then {}".format(
+            state_doc["database"],
+            datetime_to_date_str(min_datetime)
+        ))
+        alarm("asdfasdfasdkfjnasdf naskdfn aksdjfn")
+
+
+def restore(state_doc, rds_session):
+    tmp_db_description = get_database_description(
+        rds_session, state_doc["tmp_database"]
+    )
+    if tmp_db_description is None:
+        logger.info(
+            "Restoring snapshot of {database} to {tmp_database}".format(
+                **state_doc
+            )
+        )
+        transition_state(state_doc, "restore")
+        sn_ids = state_doc["database_sn_ids"]
+        if isinstance(sn_ids, basestring):
+            sn_ids = sn_ids.split(",")
+        restore_from_latest_snapshot(rds_session, state_doc["database"], sn_ids)
+    elif tmp_db_description["DBInstanceStatus"] == "available":
+        transition_state(state_doc, "modify")
+        modify(state_doc, rds_session)
+
+
+def modify(state_doc, rds_session):
+    logger.info(
+        "Modifying {tmp_database} master password and security groups".format(
+            **state_doc
+        )
+    )
+    sg_ids = state_doc["database_sg_ids"]
+    if isinstance(sg_ids, basestring):
+        sg_ids = sg_ids.split(",")
+    state_doc["tmp_password"] = modify_db_instance_for_verify(
+        rds_session, state_doc["tmp_database"], sg_ids,
+    )
+    transition_state(state_doc, "verify")
+
+
+def verify(state_doc, rds_session):
+    tmp_db_description = get_database_description(
+        rds_session, state_doc["tmp_database"]
+    )
+    tmp_db_status = tmp_db_description["DBInstanceStatus"]
+    tmp_db_event_messages = rds_event_messages(
+        rds_session, state_doc["tmp_database"]
+    )
+    if 'Reset master credentials' in tmp_db_event_messages and tmp_db_status == "available":
```

Comment:
Is this because it does no verification currently?  Basically this is as far as it goes before saying things went ok?

Reply:
Yup. The acutal verification routine (connecting to postgresql or mysql) would be triggered in this block.